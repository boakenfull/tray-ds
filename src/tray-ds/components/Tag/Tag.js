import React from "react";
import { string, bool } from "prop-types";

import { TagStyled } from "./Tag.styled";

const Tag = ({ children, isSecondary }) => (
  <TagStyled isSecondary={isSecondary}>{children}</TagStyled>
);

Tag.propTypes = {
  children: string.isRequired,
  isSecondary: bool,
};

Tag.defaultProps = {
  isSecondary: false
};

export default Tag;