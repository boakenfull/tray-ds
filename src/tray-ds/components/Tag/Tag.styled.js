import styled from "styled-components";

import { white, cornflowerBlue, frenchGray } from "../../styles/colourPalette";
import { xsmallFontSize } from "../../styles/variables";

export const TagStyled = styled.span`
  padding: 0 4px;
  font-size: ${xsmallFontSize};
  height: 16px;
  color: ${white};
  font-weight: 600;
  background-color: ${(props) => (props.isSecondary ? frenchGray : cornflowerBlue)};
  border-radius: 4px;
`;
