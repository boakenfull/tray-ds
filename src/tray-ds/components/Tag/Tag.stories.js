import React from "react";
import { text } from "@storybook/addon-knobs"

import Tag from "./";

export default { title: "Tag" };

export const Primary = () => <Tag>{text("Text", "Tag Primary")}</Tag>;
export const Secondary = () => <Tag isSecondary>{text("Text", "Tag Secondary")}</Tag>;
