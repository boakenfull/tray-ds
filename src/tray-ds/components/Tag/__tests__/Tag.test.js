import React from "react";
import { render } from "@testing-library/react";

import Tag from "..";

test("should render the tag component", () => {
  const { container } = render(<Tag>Text</Tag>);
  expect(container).toMatchSnapshot();
});

test("should render the secondary tag component", () => {
  const { container } = render(<Tag isSecondary>Text</Tag>);
  expect(container).toMatchSnapshot();
});
