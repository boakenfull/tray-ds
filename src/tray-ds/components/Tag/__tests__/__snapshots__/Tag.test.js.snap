// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the secondary tag component 1`] = `
<div>
  <span
    class="sc-AxjAm fcdCyW"
  >
    Text
  </span>
</div>
`;

exports[`should render the tag component 1`] = `
<div>
  <span
    class="sc-AxjAm yySCJ"
  >
    Text
  </span>
</div>
`;
